apps/total-typescript/src/components/exercise/video.tsx

```diff
@@ -0,0 +1,55 @@
+import {SanityDocument} from '@sanity/client'
+import {Exercise} from '../../lib/exercises'
+import * as React from 'react'
+import {useConvertkit} from '../../hooks/use-convertkit'
+import {useMuxPlayer} from '../../hooks/use-mux-player'
+import {
+  BlockedOverlay,
+  DefaultOverlay,
+  ExerciseOverlay,
+  FinishedOverlay,
+  LoadingOverlay,
+} from '../exercise-overlay'
+import cx from 'classnames'
+import MuxPlayer, {MuxPlayerProps} from '@mux/mux-player-react'
+
+type VideoProps = {
+  module: SanityDocument
+  section?: SanityDocument
+  exercise: Exercise
+  ref: any
+}
+
+export const Video: React.FC<VideoProps> = React.forwardRef(
+  ({module, exercise, section}, ref: any) => {
+    const {subscriber, loadingSubscriber} = useConvertkit()
+    const isExercise = Boolean(exercise._type === 'exercise')
+    const {muxPlayerProps, displayOverlay, nextExercise, canShowVideo} =
+      useMuxPlayer()
+
+    return (
+      <>
+        {displayOverlay && (
+          <>
+            {nextExercise ? (
+              <>{isExercise ? <ExerciseOverlay /> : <DefaultOverlay />}</>
+            ) : (
+              <FinishedOverlay />
+            )}
+          </>
+        )}
+        <div
+          className={cx('relative flex w-full items-center justify-center', {
+            hidden: displayOverlay,
+          })}
+        >
+          {canShowVideo ? (
+            <MuxPlayer ref={ref} {...(muxPlayerProps as MuxPlayerProps)} />
+          ) : (
+            <>{loadingSubscriber ? <LoadingOverlay /> : <BlockedOverlay />}</>
```

Comment:
`BlockedOverlay` needs internal logic to understand where/why it is being show (`moduleType`)

`loadingUser` will need to come from the context as well